components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/consent/ConsentMgtPostAuthnHandler.java

```diff
@@ -488,7 +335,7 @@ private UserConsent processUserConsent(HttpServletRequest request, Authenticatio
                                                                           approvedClamMetaData);
 
         if (isMandatoryClaimsDisapproved(consentClaimsData.getMandatoryClaims(), disapprovedClaims)) {
-            throw new PostAuthenticationFailedException("Consent Denied for Mandatory Attributes",
+            throw new PostAuthenticationFailedException("Consent Denied for Mandatory Attributes.",
```

Comment:
Shall we add the message Authentication failed as well. Otherwise user may try to go back from browswer and retry to give consent in the same flow.